sernet.gs.ui.rcp.main/src/sernet/verinice/iso27k/service/CopyService.java

```diff
@@ -98,13 +100,21 @@ public void run()  {
             numberOfElements = uuidList.size();
             // -1 means unknown runtime
             progressObserver.beginTask(Messages.getString("CopyService.1", numberOfElements), -1); //$NON-NLS-1$
-            CopyCommand cc = new CopyCommand(this.selectedGroup.getUuid(), uuidList, getPostProcessorList(), this.copyLinks);
-            cc.setCopyAttachments(isCopyAttachments());
-            cc = getCommandService().executeCommand(cc);
-            numberOfElements = cc.getNumber();
+            if (isModelingTemplateActive()) {
+                sernet.verinice.service.commands.templates.CopyCommand cc = new sernet.verinice.service.commands.templates.CopyCommand(this.selectedGroup.getUuid(), uuidList, getPostProcessorList(), this.copyLinks);
+                cc.setCopyAttachments(isCopyAttachments());
+                cc = getCommandService().executeCommand(cc);
+                numberOfElements = cc.getNumber();
+                newElements = cc.getNewElements();
+            } else {
+                sernet.verinice.service.commands.CopyCommand cc = new sernet.verinice.service.commands.CopyCommand(this.selectedGroup.getUuid(), uuidList, getPostProcessorList(), this.copyLinks);
+                cc.setCopyAttachments(isCopyAttachments());
+                cc = getCommandService().executeCommand(cc);
+                numberOfElements = cc.getNumber();
+                newElements = cc.getNewElements();
+            }
```

Comment:
The way I see it you do exactly the same thing with both `CopyCommand`s, i.e. you can move most of the code within the if scope:
```java
if (...) {
  cc = new ..
} else {
  cc = new ...
}
cc.setCopy...
...
```

Reply:
For then you have to change GenericCommand or ICommand first.